src/main/java/com/aws/greengrass/shadowmanager/model/configuration/ShadowSyncConfiguration.java

```diff
@@ -261,6 +274,88 @@ private static void processThingShadowSyncConfiguration(Set<ThingShadowSyncConfi
+        }
+        Map<String, Object> configMap = (Map<String, Object>) configObject;
+
+        Object enabledObject = configMap.get(CONFIGURATION_ADD_ON_INTERACTION_ENABLED_TOPIC);
+        if (enabledObject == null) {
+            return;
+        }
+        if (!(enabledObject instanceof Boolean)) {
+            throw unexpectedTypeException(configObject, String.format(
+                "%s.%s", CONFIGURATION_ADD_ON_INTERACTION_TOPIC, CONFIGURATION_ADD_ON_INTERACTION_ENABLED_TOPIC
+            ));
+        }
+        addOnInteraction.setEnabled((Boolean) enabledObject);
+    }
+
+    private static void processAddedOnInteractionConfigurations(
+        Map<String, Object> configTopicsPojo, Set<ThingShadowSyncConfiguration> syncConfigurationSet
+    ) {
+        Object configObject = configTopicsPojo.get(CONFIGURATION_ADDED_ON_INTERACTION_TOPIC);
+        if (configObject == null) {
+            return;
+        }
+        if (!(configObject instanceof Map)) {
+            throw unexpectedTypeException(configObject, CONFIGURATION_ADDED_ON_INTERACTION_TOPIC);
+        }
+        Map<String, Object> configMap = (Map<String, Object>) configObject;
+
+        Set<ThingShadowSyncConfiguration> thingShadowsAddedOnInteraction = new HashSet<>();
+        configMap.forEach((thingName, shadowsConfigObject) -> processAddedOnInteractionThingConfiguration(
+            thingName, shadowsConfigObject, thingShadowsAddedOnInteraction
+        ));
+
+        // merge shadows added on interaction and not on interaction
+        for (ThingShadowSyncConfiguration thingShadow: thingShadowsAddedOnInteraction) {
+            if (syncConfigurationSet.contains(thingShadow)) {
+                continue;
+            }
+            syncConfigurationSet.add(thingShadow);
+        }
+    }
+
+    private static void processAddedOnInteractionThingConfiguration(
+        String thingName, Object configObject, Set<ThingShadowSyncConfiguration> syncConfigurationSet
+    ) {
+        if (configObject == null) {
+            return;
+        }
+        if (!(configObject instanceof Map)) {
+            throw unexpectedTypeException(configObject, String.format(
+                "%s[%s]", CONFIGURATION_ADDED_ON_INTERACTION_TOPIC, thingName
+            ));
+        }
+
+        Map<String, Boolean> configMap = (Map<String, Boolean>)configObject;
+        configMap.forEach((shadowName, onInteraction) -> {
+            if (!onInteraction) {
+                return;
+            }
+            ThingShadowSyncConfiguration sync = ThingShadowSyncConfiguration
+                .builder()
+                .thingName(thingName)
+                .shadowName(shadowName)
+                .addedOnInteraction(true)
+                .build();
+            syncConfigurationSet.add(sync);
+            if (syncConfigurationSet.contains(sync)) {
+                // it already contains this sync configuration and it was not added on interaction.
```

Comment:
Since `sync`created in L344 is always going to have `addedOnInteraction` as true and if it already exists in the `syncConfigurationSet`, it should mean that it was added on interaction.